packages/nitrogen/src/views/CppHybridViewComponent.ts

```diff
@@ -75,9 +75,27 @@ export function createViewComponentShadowNodeFiles(
     (p) => `CachedProp<${p.type.getCode('c++')}> ${escapeCppName(p.name)};`
   )
   const cases = props.map((p) => `case hashString("${p.name}"): return true;`)
-  const includes = props.flatMap((p) =>
-    p.getRequiredImports('c++').map((i) => includeHeader(i, true))
-  )
+  // Collect and dedupe dynamic includes, excluding ones already added statically above
+  const fixedHeaderIncludes = new Set<string>([
+    '#include <optional>',
+    '#include <NitroModules/NitroDefines.hpp>',
+    '#include <NitroModules/NitroHash.hpp>',
+    '#include <NitroModules/CachedProp.hpp>',
+    '#include <react/renderer/core/ConcreteComponentDescriptor.h>',
+    '#include <react/renderer/core/PropsParserContext.h>',
+    '#include <react/renderer/components/view/ConcreteViewShadowNode.h>',
+    '#include <react/renderer/components/view/ViewProps.h>',
+  ])
+  const dynamicIncludesSet = new Set<string>()
+  for (const prop of props) {
+    for (const import_ of prop.getRequiredImports('c++')) {
```

Comment:
`import_` -> name with a `_` suffix is not clean imo. I'd name it something different, like `requiredImport`.